Mathlib/RingTheory/FractionalIdeal/Basic.lean

```diff
@@ -134,6 +134,40 @@ protected theorem isFractional (I : FractionalIdeal S P) : IsFractional S (I : S
   I.prop
 #align fractional_ideal.is_fractional FractionalIdeal.isFractional
 
+/-- An element of `S` such that `I.den • I.num = I`, see `FractionalIdeal.num` and
+`FractionalIdeal.den_mul_eq_num`. -/
+noncomputable def den (I : FractionalIdeal S P) : S :=
+  ⟨I.2.choose, I.2.choose_spec.1⟩
+
+/-- An ideal of `R` such that `I.den • I.num = I`, see `FractionalIdeal.den` and
+`FractionalIdeal.den_mul_eq_num`. -/
+noncomputable def num (I : FractionalIdeal S P) : Ideal R :=
+  (I.den • (I : Submodule R P)).comap (Algebra.linearMap R P)
+
+theorem den_mul_eq_num (I : FractionalIdeal S P) :
+    I.den • (I : Submodule R P) = Submodule.map (Algebra.linearMap R P) I.num := by
+  rw [den, num, Submodule.map_comap_eq]
+  refine (inf_of_le_right ?_).symm
+  rintro _ ⟨a, ha, rfl⟩
+  exact I.2.choose_spec.2 a ha
+
+/-- The linear equivalence between the fractional ideal `I` and the integral ideal `I.num`
+defined by mapping `x` to `den I • x`. -/
+noncomputable def equivNum [NoZeroSMulDivisors R P]
+    (h_inj : Function.Injective (algebraMap R P)) {I : FractionalIdeal S P}
```

Comment:
Isn't `h_inj` automatic since you assume `NoZeroSMulDivisors R P`?

Similarly below.

Reply:
Indeed, I just need to add a `Nontrivial` hypothesis instead.